webrtc/RTCPeerConnection-helper.js

```diff
@@ -159,19 +159,17 @@ function test_never_resolve(testFunc, testName) {
   }, testName);
 }
 
-// Helper function to exchange ice candidates between
-// two local peer connections
-function exchangeIceCandidates(pc1, pc2) {
+// Helper function to exchange ice candidates between two local peer connections.
+// Accepts an optional handshakePromise to wait for setRemoteDescription() to be
+// done before calling addIceCandidate().
+function exchangeIceCandidates(pc1, pc2, handshakePromise) {
```

Comment:
I don't think we should land this. This is queuing ICE candidates again, like in https://github.com/web-platform-tests/wpt/pull/14417#discussion_r240668244.

It also doesn't appear to fix the problem.

As long as people call *exchangeIceCandidates(pc1, pc2)* before *doSignalingHandshake(pc1, pc2)* there should be no race. Let's discuss in https://github.com/web-platform-tests/wpt/issues/12028#issuecomment-448015553.